### PR TITLE
Hide time inputs in the date picker when the column does not support date+time filters

### PR DESCRIPTION
**Pull request by ranquild**
Related to https://github.com/metabase/metabase/issues/48608

Takes `availableUnits` into account in specific & relative date pickers.

How to verify:
- New -> Question -> People
- Filter -> Created At -> Specific dates... -> There is "Add time" button
- Filter -> Birth Date -> Specific dates... -> There is no "Add time" button
- Filter -> Birth Date -> Relative dates... -> There is no "hours" unit

Screenshot 2024-10-16 at 23 16 56
Screenshot 2024-10-16 at 23 17 02

### CODE REVIEW

File: frontend/src/metabase/querying/filters/components/FilterModal/DateFilterEditor/DateFilterEditor.unit.spec.tsx [hunk 1]
```diff
@@ -46,162 +46,188 @@ describe("DateFilterEditor", () => {
   const findColumn = columnFinder(defaultQuery, availableColumns);
   const column = findColumn("ORDERS", "CREATED_AT");
 
-  describe("new filter", () => {
-    it("should add a relative date filter from a shortcut", async () => {
```
Comment (ranquild) [inline frontend/src/metabase/querying/filters/components/FilterModal/DateFilterEditor/DateFilterEditor.unit.spec.tsx:50]: whitespace

**Comment by kamilmielnik** [comment]
> @[kamilmielnik](https://github.com/kamilmielnik) kamilmielnik [force-pushed](https://github.com/metabase/metabase/compare/5d4388a2d9dd8bda5f20a40dcc6afc657e3182a4..6631dc1c0dd723eda978f4890bae671ce6a17b20) the 48608-date-filter-time branch from [5d4388a ](https://github.com/metabase/metabase/commit/5d4388a2d9dd8bda5f20a40dcc6afc657e3182a4)to [6631dc1 ](https://github.com/metabase/metabase/commit/6631dc1c0dd723eda978f4890bae671ce6a17b20)
[now](https://github.com/metabase/metabase/pull/48818#event-14717662196)

I accidentally pushed my commits to the wrong branch :see_no_evil: Fixed.

File: frontend/src/metabase/querying/filters/components/DatePicker/DatePicker.tsx [hunk 2]
```diff
@@ -22,9 +22,10 @@ interface DatePickerProps {
   availableOperators?: ReadonlyArray<DatePickerOperator>;
   availableShortcuts?: ReadonlyArray<DatePickerShortcut>;
   availableUnits?: ReadonlyArray<DatePickerExtractionUnit>;
-  canUseRelativeOffsets?: boolean;
-  backButton?: ReactNode;
   isNew?: boolean;
+  canSetTime?: boolean;
```
[the remark this answers is not in the document]
Comment (ranquild) [reply frontend/src/metabase/querying/filters/components/DatePicker/DatePicker.tsx]: Fixed

File: frontend/src/metabase/querying/filters/components/FilterModal/DateFilterEditor/DateFilterEditor.unit.spec.tsx [hunk 3]
```diff
@@ -46,162 +46,184 @@ describe("DateFilterEditor", () => {
...
-        filter,
-      });
-      expect(screen.getByText("Excludes 5 PM")).toBeInTheDocument();
+    await userEvent.click(screen.getByLabelText("Clear"));
+    expect(getNextFilterName()).toBe(null);
+  });
 
-      await userEvent.click(screen.getByLabelText("Clear"));
-      expect(getNextFilterName()).toBe(null);
+  it("should not allow to set time for a date only column", async () => {
```
Comment (ranquild) [inline frontend/src/metabase/querying/filters/components/FilterModal/DateFilterEditor/DateFilterEditor.unit.spec.tsx:205]: New test

**Review by kamilmielnik — APPROVED** [review]

**Pull request merged**